src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,152 @@
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractInternalsVisibleToCompletionProvider : CommonCompletionProvider
+    {
+        private const string ProjectGuidKey = nameof(ProjectGuidKey);
+
+        protected abstract bool IsPositionEntirelyWithinStringLiteral(SyntaxTree syntaxTree, int position, CancellationToken cancellationToken);
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '\"';
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var syntaxTree = await context.Document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            if (IsPositionEntirelyWithinStringLiteral(syntaxTree, context.Position, cancellationToken))
+            {
+                var syntaxFactsService = context.Document.GetLanguageService<ISyntaxFactsService>();
+                var token = syntaxTree.FindTokenOnLeftOfPosition(context.Position, cancellationToken);
+                var attributeSyntaxNode = GetAttributeSyntaxNodeOfToken(syntaxFactsService, token);
+                if (attributeSyntaxNode == null)
+                {
+                    return;
+                }
+
+                if (await CheckTypeInfoOfAttributeAsync(context, attributeSyntaxNode).ConfigureAwait(false))
+                {
+                    AddAssemblyCompletionItems(context, cancellationToken);
+                }
+            }
+        }
+
+        private static SyntaxNode GetAttributeSyntaxNodeOfToken(ISyntaxFactsService syntaxFactsService, SyntaxToken token)
+        {
+            //Supported cases:
+            //[Attribute("|
+            //[Attribute(parameterName:"Text|")
+            //Also supported but excluded by IsPositionEntirelyWithinStringLiteral in ProvideCompletionsAsync
+            //[Attribute(""|
+            //[Attribute("Text"|)
+            var node = token.Parent;
+            if (syntaxFactsService.IsStringLiteralExpression(node))
+            {
+                // Edge case: ElementAccessExpressionSyntax is present if the following statement is another attribute:
+                //   [assembly: System.Runtime.CompilerServices.InternalsVisibleTo("|
+                //   [assembly: System.Reflection.AssemblyVersion("1.0.0.0")]
+                //   [assembly: System.Reflection.AssemblyCompany("Test")]
+                while (syntaxFactsService.IsElementAccessExpression(node.Parent))
```

Comment:
`IsElementAccessExpression` is called after the check `IsPositionEntirelyWithinStringLiteral` and `IsStringLiteralExpression` so this should not trigger anything. See also the [CodeCompletionContainsIsEmptyUntilDoubleQuotesAreEntered](https://github.com/MaStr11/roslyn/blob/9c3b1a52c198bc622eb3372341efafb2ff27c227/src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests_InternalsVisibleTo.vb#L64) test method (completion list is emtpy after [ is entered). Does this answer your question? Or should I add some more tests?